src/posit/connect/resources.py

```diff
@@ -89,38 +93,50 @@ def update(self, **attributes):  # type: ignore[reportIncompatibleMethodOverride
         super().update(**result)
 
 
-T = TypeVar("T", bound=Resource)
+_T = TypeVar("_T", bound=Resource)
+_T_co = TypeVar("_T_co", bound=Resource, covariant=True)
+
+
+class ResourceFactory(Protocol[_T_co]):
+    def __call__(self, ctx: Context, path: str, **attributes: Any) -> _T_co: ...
 
 
-class ResourceSequence(Protocol[T]):
+class ResourceSequence(Protocol[_T]):
     @overload
-    def __getitem__(self, index: SupportsIndex, /) -> T: ...
+    def __getitem__(self, index: SupportsIndex, /) -> _T: ...
 
     @overload
-    def __getitem__(self, index: slice, /) -> List[T]: ...
+    def __getitem__(self, index: slice, /) -> List[_T]: ...
 
     def __len__(self) -> int: ...
 
-    def __iter__(self) -> Iterator[T]: ...
+    def __iter__(self) -> Iterator[_T]: ...
 
     def __str__(self) -> str: ...
 
     def __repr__(self) -> str: ...
 
 
-class _ResourceSequence(Sequence[T], ResourceSequence[T]):
-    def __init__(self, ctx: Context, path: str, *, uid: str = "guid"):
+class _ResourceSequence(Sequence[_T], ResourceSequence[_T]):
+    def __init__(
+        self,
+        ctx: Context,
+        path: str,
+        factory: ResourceFactory[_T_co] = _Resource,
```

Comment:
My preference is to put uncommon args that have defaults behind some sort of "name requirement" mechanism.

```suggestion
        *,
        factory: ResourceFactory[_T_co] = _Resource,
```